includes/blocks/class-llms-blocks-course-progress-block.php

```diff
@@ -75,18 +83,22 @@ public function output( $attributes = array() ) {
 		 * Filters the block html
 		 *
 		 * @since 1.9.0
+		 * @deprecated [version]
 		 *
-		 * @param string                            $block_content The block's html.
-		 * @param array                             $attributes    The block's array of attributes.
-		 * @param LLMS_Blocks_Course_Progress_Block $block         This block object.
+		 * @param string $block_content The block's html.
+		 * @param array  $attributes    The block's array of attributes.
+		 * @param self   $block         This block object.
 		 */
-		$block_content = apply_filters( 'llms_blocks_render_course_progress_block', $block_content, $attributes, $this );
+		$block_content = apply_filters_deprecated(
+			'llms_blocks_render_course_progress_block',
+			array( $block_content, $attributes, $this ),
+			'[version]',
+			'render_block_llms/course-progress'
+		);
 
 		if ( $block_content ) {
 			echo $block_content;
 		}
 
 	}
 }
-
```

Comment:
we need to add all the public and protected methods inherited from the abstract class and deprecate them like we did with the syllabus.